src/apps/viewer/extensions/cellpack/model.ts

```diff
@@ -237,14 +264,53 @@ async function getIngredientStructure(ingredient: Ingredient, baseUrl: string) {
 
     if (source.pdb === 'None') return
 
-    const model = await getModel(source.pdb || name, baseUrl)
+    const model = await getModel(source.pdb || name,source, baseUrl)
     if (!model) return
 
     if (nbCurve) {
-        return getCurve(name, getCurveTransforms(ingredient), model)
+        return getCurve(name,ingredient, getCurveTransforms(ingredient), model)
     } else {
-        const structure = await getStructure(model, { assembly: source.biomt ? '1' : undefined })
```

Comment:
is biomt still needed for legacy models?

Reply:
yes, but I will try to redo the other models so that they are consistent and we can clean up this backward compatile code